homeassistant/components/samsungtv/config_flow.py

```diff
@@ -115,48 +133,73 @@ async def _async_set_unique_id_from_udn(
     ) -> None:
         """Set the unique id from the udn."""
         assert self._host is not None
-        await self.async_set_unique_id(self._udn, raise_on_progress=raise_on_progress)
+        # Set the unique id without raising on progress in case
+        # there are two SSDP flows with for each ST
+        await self.async_set_unique_id(self._udn, raise_on_progress=False)
         if (
             entry := self._async_update_existing_matching_entry()
-        ) and _entry_is_complete(entry):
+        ) and _entry_is_complete(entry, self._ssdp_location):
             raise data_entry_flow.AbortFlow("already_configured")
+        # Now that we have updated the config entry, we can raise
+        # if another one is progressing
+        if raise_on_progress:
+            await self.async_set_unique_id(self._udn, raise_on_progress=True)
 
     def _async_update_and_abort_for_matching_unique_id(self) -> None:
         """Abort and update host and mac if we have it."""
         updates = {CONF_HOST: self._host}
         if self._mac:
             updates[CONF_MAC] = self._mac
+        if self._ssdp_location:
+            updates[CONF_SSDP_LOCATION] = self._ssdp_location
         self._abort_if_unique_id_configured(updates=updates)
 
     async def _try_connect(self) -> None:
         """Try to connect and check auth."""
-        for method in SUPPORTED_METHODS:
-            self._bridge = SamsungTVBridge.get_bridge(self.hass, method, self._host)
-            result = await self._bridge.async_try_connect()
-            if result == RESULT_SUCCESS:
-                return
-            if result != RESULT_CANNOT_CONNECT:
-                raise data_entry_flow.AbortFlow(result)
-        LOGGER.debug("No working config found")
-        raise data_entry_flow.AbortFlow(RESULT_CANNOT_CONNECT)
+        result, method, _info = await self._async_get_device_info_and_method()
+        if result != RESULT_SUCCESS:
+            LOGGER.debug("No working config found for %s", self._host)
+            raise data_entry_flow.AbortFlow(result)
+        assert method is not None
+        LOGGER.debug("Try connect determined method to use: %s", method)
+        self._bridge = SamsungTVBridge.get_bridge(self.hass, method, self._host)
+        if self._bridge.method != METHOD_WEBSOCKET:
```

Comment:
I still don't like this since it only gives you one chance to hit the button on the websocket tv